codespan-reporting/src/term/views.rs

```diff
@@ -200,24 +211,35 @@ where
                         _ => (label_index, label.style, MultiLabel::Top(..label_start)),
                     });
 
+                // The first line has to be rendered so the start of the label is visible.
+                start_line.omittable = false;
+
                 // Marked lines
                 //
                 // ```text
                 // 5 │ │     0 0 => "FizzBuzz"
                 // 6 │ │     0 _ => "Fizz"
                 // 7 │ │     _ 0 => "Buzz"
                 // ```
-                // TODO(#125): If start line and end line are too far apart, add a source break.
                 for line_index in (start_line_index + 1)..end_line_index {
                     let line_range = files.line_range(label.file_id, line_index).unwrap();
                     let line_number = files.line_number(label.file_id, line_index).unwrap();
 
                     outer_padding = std::cmp::max(outer_padding, count_digits(line_number));
 
-                    labeled_file
-                        .get_or_insert_line(line_index, line_range, line_number)
-                        .multi_labels
+                    let line = labeled_file.get_or_insert_line(line_index, line_range, line_number);
+
+                    line.multi_labels
                         .push((label_index, label.style, MultiLabel::Left));
+
+                    // The line should be rendered to match the configuration of how much context to show.
+                    line.omittable &= !(
```

Comment:
I'm wondering if this would be clearer if the flag was reversed, eg. calling it `must_render` or something 🤔